mmdet/core/evaluation/eval_hooks.py

```diff
@@ -21,8 +22,8 @@ def __init__(self, dataset, interval=1):
         if isinstance(dataset, Dataset):
             self.dataset = dataset
         elif isinstance(dataset, dict):
-            self.dataset = obj_from_dict(dataset, datasets,
-                                         {'test_mode': True})
+            self.dataset = build_from_cfg(dataset, datasets.DATASETS,
```

Comment:
We can use [`build_dataset()`](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/builder.py#L30) here.

Reply:
Yes that's probably best. How can we ensure that test_mode = True when using build_dataset()? It won't accept default_args. Should it be changed too?

Reply:
Yes you can add an argument for `build_dataset`.

Reply:
Ok done.